src/adapter/src/coord.rs

```diff
@@ -779,6 +779,7 @@ pub(crate) enum StageResult<T> {
 }
 
 /// Common functionality for [Coordinator::sequence_staged].
+#[async_trait::async_trait(?Send)]
```

Comment:
nit: you should be able to remove the `Box<...>` wrapper from `async fn stage(...)` now and change it to just `Self`

Reply:
This complains with


```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> src/adapter/src/coord.rs:791:17
    |
791 |     ) -> Result<StageResult<Self>, AdapterError>;
    |                 ^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
note: required by a bound in `StageResult`
   --> src/adapter/src/coord.rs:774:29
    |
774 | pub(crate) enum StageResult<T> {
    |                             ^ required by this bound in `StageResult`
help: consider relaxing the implicit `Sized` restriction
    |
774 | pub(crate) enum StageResult<T: ?Sized> {
    |                              ++++++++
```

So I'm going to merge as is and maybe we can improve it later.